containers/containers.go

```diff
@@ -12,15 +12,51 @@ import (
 //
 // The resources specified in this object are used to create tasks from the container.
 type Container struct {
-	ID          string
-	Labels      map[string]string
-	Image       string
-	Runtime     RuntimeInfo
-	Spec        *types.Any
-	RootFS      string
+	// ID uniquely identifies the container in a nameapace.
+	//
+	// This property is required and cannot be changed after creation.
+	ID string
+
+	// Labels provide metadata extension for a contaienr.
+	//
+	// These are optional and fully mutable.
+	Labels map[string]string
+
+	// Image specifies the image refenence used for a container.
+	//
+	// This property is optional but immutable.
+	Image string
+
+	// Runtime specifies which runtime should be used when launching container
+	// tasks.
+	//
+	// This property is required and immutable.
+	Runtime RuntimeInfo
+
+	// Spec should carry the the tuntime specification used to implement the
+	// container.
+	//
+	// This field is required but mutable.
+	Spec *types.Any
+
+	// RootFS specifies the snapshot key to use for the container's root
+	// filesystem. When starting a task from this container, a caller should
+	// look up the mounts from the snapshot service and include those on the
+	// task create request.
+	//
+	// This field is not required but immutable.
+	RootFS string
```

Comment:
I'm not quite sure. It might be interesting to have `RootFS` only be a snapshotter key if `Snapshotter` is set. If `Snapshotter` is not set, it could be a filesystem path. There are lot of benefits to that model.

cc @dmcgowan @AkihiroSuda